File: 2015/03/part2.rb
require 'set'

hash = Hash.new { |h, k| h[k] = Set.new }
hash[0] << 0
sx, sy = 0, 0
rx, ry = 0, 0

File.read('./input.txt').split('').each_with_index do |dir, index|
  dx, dy = 0, 0

  case dir
    when '<'
      dx = -1
    when '>'
      dx = 1
    when '^'
      dy = 1
    when 'v'
      dy = -1
  end

  if index % 2 == 0
    rx += dx
    ry += dy
    hash[rx] << ry
  else
    sx += dx
    sy += dy
    hash[sx] << sy
  end

end

puts hash.values.map(&:size).inject(:+)

File: 2015/01/part2.ex
defmodule AdventOfCode.Day1 do

  def solve(str) do
    str |> String.graphemes |> Enum.map(&brace_to_int/1) |> find_basement(0, 0)
  end

  defp find_basement(arr, counter, index) do
    if counter < 0 || index == Enum.count(arr) - 1 do
      index
    else
      find_basement(arr, counter + Enum.at(arr,index), index + 1)
    end
  end

  defp brace_to_int(x) when x == "(" do 1 end
  defp brace_to_int(_) do  -1 end

end

{ :ok, str } = File.read('./input.txt')
IO.puts AdventOfCode.Day1.solve(str)